lib/functoria/tool.ml

```diff
@@ -197,17 +197,104 @@ module Make (P : S) = struct
     rm_gen_files ()
 
   (* App builder configuration *)
-  let with_alias ~save_args args ~depext:_ ~extra_repo:_ ?ppf ?err_ppf argv =
+  let configure ({ args; _ } : _ Cli.configure_args) ?ppf ?err_ppf argv =
+    let file = args.Cli.config_file in
+    let* () =
+      let* is_file = Action.is_file file in
+      if not is_file then
+        Action.errorf "configuration file %a missing" Fpath.pp file
+      else Action.ok ()
+    in
+    let* () =
+      let* data =
+        let cmd = Bos.Cmd.(v "head" % "-1" % p file) in
+        Action.run_cmd_out ~err:`Null cmd
+      in
+      let pkg = "%%NAME%%" in
+      let first_str = "(* " ^ pkg in
```

Comment:
Do you mind extracting this function (and exposing it somewhere) in the API and adding a few unit tests? I understand testing is annoying because of the `VERSION_NUM` substitution, but I can see becoming it hard to maintain without at least a few simple tests.

Reply:
(we should probably have a way to inject `VERSION_NUM` in our tests, but that is more complicated that just unit-testing this function and can definitely come in a separate PR if their is a need for better tests later)

Reply:
that's all now done in [9b667fe](https://github.com/mirage/mirage/pull/1466/commits/9b667fe62f1ae54fc1b0aa6ea32c886f39f049a4) and [cfbed95](https://github.com/mirage/mirage/pull/1466/commits/cfbed95d98af62d1f6f542da51d12c60b32bea60)

Your feedback would be valuable (or a merge :)